heaps/min_heap.py

```diff
@@ -19,19 +19,38 @@ def __init__(self):
     def add(self, key, value = None):
```

Comment:
👀 Very  nice! however space complexity would be O(log n) because of `heap_up`'s recursive call stack